analyzers/tests/SonarAnalyzer.UnitTest/Common/BidirectionalDictionaryTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+
+namespace SonarAnalyzer.UnitTest.Common
+{
+    [TestClass]
+    public class BidirectionalDictionaryTests
+    {
+        private BidirectionalDictionary<int, int> sut;
+
+        [TestInitialize]
+        public void Initialize() =>
+            sut = new BidirectionalDictionary<int, int>();
+
+        [TestMethod]
+        public void AddKeyA_AlreadyExists_Throws()
+        {
+            sut.Add(1, 2);
+            sut.Invoking(x => x.Add(1, 3)).Should().Throw<ArgumentException>().WithMessage("An element with the same key already exists in the BidirectionalDictionary.");
+        }
+
+        [TestMethod]
+        public void AddKeyB_AlreadyExists_Throws()
+        {
+            sut.Add(1, 2);
+            sut.Invoking(x => x.Add(3, 2)).Should().Throw<ArgumentException>().WithMessage("An element with the same key already exists in the BidirectionalDictionary.");
+        }
+
+        [TestMethod]
+        public void Keys_ReturnsCorrectKeys()
+        {
+            sut.Add(1, 11);
+            sut.Add(2, 12);
+            sut.Add(3, 13);
+            sut.Add(4, 14);
+            sut.Add(5, 15);
+
+            sut.AKeys.Should().OnlyContain(x => x == 1 || x == 2 || x == 3 || x == 4 || x == 5);
+            sut.BKeys.Should().OnlyContain(x => x == 11 || x == 12 || x == 13 || x == 14 || x == 15);
```

Comment:
We should use `BeEquivalentTo`, since it's meant specifically for comparing collections.
```suggestion
            sut.AKeys.Should().BeEquivalentTo(new[] {1, 2, 3, 4, 5});
            sut.BKeys.Should().BeEquivalentTo(new[] {11, 12, 13, 14, 15});
```